database/repository/header_repository.go

```diff
@@ -133,6 +133,41 @@ func NewHeadersRepository(db *sql.HeadersDb) *HeaderRepository {
 	return &HeaderRepository{db: db}
 }
 
+func (r *HeaderRepository) GetMerkleRoots(batchSize int, lastEvaluatedKey int) (*domains.MerkleRootsESKPagedResponse, error) {
+	orderByField := "BlockHeight"
+	sortDirection := "ASC"
+	merklerootsFromDb, err1 := r.db.GetMerkleRoots(batchSize, lastEvaluatedKey)
+	merklerootsTopHeight, err2 := r.GetHeadersCount()
+
+	if err1 != nil {
+		return nil, err1
+	}
+
+	if err2 != nil {
+		return nil, err2
+	}
```

Comment:
It's rather a good practice to return an error right after it occurs.
```suggestion
	merklerootsFromDb, err := r.db.GetMerkleRoots(batchSize, lastEvaluatedKey)
	if err != nil {
		return nil, err
	}
	merklerootsTopHeight, err := r.GetHeadersCount()
	if err != nil {
		return nil, err
	}
```

Reply:
I see @wregulski was faster :D - so I'm just resolving this comment 🙈